pkg/sidecarlogresults/sidecarlogresults.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sidecarlogresults
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	resultSeparator = " "
+	resultFinisher  = "---"
```

Comment:
Trying to understand the rationale behind using this kind pattern for the finisher? Are we trying to model it after `yaml`?

Reply:
It could really be anything. I needed something to separate result names and values (I ended up using a whitespace but it could be anything). I also needed something to separate results. I ended up using `---` between new line as a separator between two results (yes, motivated by `yaml`).

Reply:
Based on @skaegi suggestion, I'me removing these constants and instead using json Marshal to encode the strings so while reading the logs, each result will be one line. That will remove this unnecessary complexity.

Reply:
I removed this dependency and updated the code. Please let me know if you have additional concerns.